File: SDKs/spotify.com/2021.8.15/python/src/sdk/models/shared/audiofeaturesobject.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class AudioFeaturesObject:
    r"""AudioFeaturesObject

    https://developer.spotify.com/documentation/web-api/reference/#object-audiofeaturesobject - Find more info on the official Spotify Web API Reference
    """
    
    acousticness: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('acousticness') }})
    analysis_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('analysis_url') }})
    danceability: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('danceability') }})
    duration_ms: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('duration_ms') }})
    energy: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('energy') }})
    id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    instrumentalness: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('instrumentalness') }})
    key: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('key') }})
    liveness: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('liveness') }})
    loudness: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('loudness') }})
    mode: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('mode') }})
    speechiness: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('speechiness') }})
    tempo: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tempo') }})
    time_signature: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('time_signature') }})
    track_href: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('track_href') }})
    type: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    uri: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('uri') }})
    valence: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('valence') }})
    

File: SDKs/spotify.com/2021.8.15/typescript/src/sdk/models/shared/audiofeaturesobject.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// AudioFeaturesObject
/** 
 * https://developer.spotify.com/documentation/web-api/reference/#object-audiofeaturesobject - Find more info on the official Spotify Web API Reference
**/
export class AudioFeaturesObject extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=acousticness" })
  acousticness?: number;

  @SpeakeasyMetadata({ data: "json, name=analysis_url" })
  analysisUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=danceability" })
  danceability?: number;

  @SpeakeasyMetadata({ data: "json, name=duration_ms" })
  durationMs?: number;

  @SpeakeasyMetadata({ data: "json, name=energy" })
  energy?: number;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=instrumentalness" })
  instrumentalness?: number;

  @SpeakeasyMetadata({ data: "json, name=key" })
  key?: number;

  @SpeakeasyMetadata({ data: "json, name=liveness" })
  liveness?: number;

  @SpeakeasyMetadata({ data: "json, name=loudness" })
  loudness?: number;

  @SpeakeasyMetadata({ data: "json, name=mode" })
  mode?: number;

  @SpeakeasyMetadata({ data: "json, name=speechiness" })
  speechiness?: number;

  @SpeakeasyMetadata({ data: "json, name=tempo" })
  tempo?: number;

  @SpeakeasyMetadata({ data: "json, name=time_signature" })
  timeSignature?: number;

  @SpeakeasyMetadata({ data: "json, name=track_href" })
  trackHref?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: string;

  @SpeakeasyMetadata({ data: "json, name=uri" })
  uri?: string;

  @SpeakeasyMetadata({ data: "json, name=valence" })
  valence?: number;
}
